juris-ca.csl

```diff
@@ -35,7 +35,7 @@
       <else-if type="report">
         <names variable="authority"/>
       </else-if>
-      <else-if type="regulation" variable="authority" match="all">
+      <else-if type="regulation" variable="authority genre volume" match="all">
```

Comment:
What happens in the case that a regulation has an authority but no genre/volume? Is the result correct?

What about regulation with authority and genre? Shouldn't the genre be given even in absence of volume information?

Reply:
As I said in another comment, this is not the general "regulation" item, but rather, municipal bylaws, which don't have their own item types. I haven't had a chance to do standard regulations yet (because I'll have to create style-modules for each subjurisdiction to do it).

There should never be a case where the "genre" "volume" and "authority" fields are given. In this context, "authority" = the city, "genre" = revised or unrevised by-law, and "volume" = bylaw-number or bylaw chapter number. 

There should never be a case where all of these aren't present. If any of these fields are empty, the citation will be incorrect, due to missing information.

Reply:
How are you determining that the item is a municipal by-law and not another kind of regulation?

Reply:
Right now, that is done using the authority value. 

In Canadian legal citation, regulation citations don't refer to the enabling statute, which means that the "authority" variable is empty. 

When it comes to municipal bylaws, the authority variable is used to input the city in question. Municipal bylaws do not use the "container-title" variable, which is used by the regular regulations. 

So right now : 
Volume is always used, both by regulations and bylaws
Genre is always used by bylaws, and only sometimes used by regulations.
Authority is always used by bylaws, but never by regulations.
Container-title is always used by regulations, but never by bylaws. 

However, if you want, in order to guarantee correct rendering if someone mistakenly puts the enabling statute in the "authority" field, I can make sure that when regular regulations are implemented, they check for the "container-title" variable, and only if isn't present would a check for authority take place. That would ensure that both types will always render correctly.

Reply:
@georgd Could you comment here? This set of conditions seems rather fragile, but my impression might be off.

Reply:
That’s indeed fragile but I don’t see a better way to do this, given @Droitslinguistiques' explanation. I‘ve been in that situation too, not only once. The most important thing here is good documentation for the end users, I’d say.

Reply:
We're working on making a complete document that will outline what fields to fill for each item type. Progress is slow, because I keep having to find workarounds to account for citation types which don't have a dedicated item type (such as municipal regulations). 

To my understanding, since it is impossible in CSL to check the content of a variable, only whether it exists or not, workarounds like this are our only option.